plutus-chain-index/src/Marconi/Index/Datum.hs

```diff
@@ -72,6 +72,12 @@ open dbPath (Depth k) = do
 
 -- | This function is used to query the data stored in the indexer as a whole:
 --   data that can still change (through rollbacks), buffered data and stored data.
+--
+--   Here is a query that takes into account all received events:
+--   > getEvents (ix ^. storage) >>= query ix <hash>
+--
+--   Here is a query that only takes into account stored events:
+--   > query ix <hash> []
```

Comment:
I think the examples got mixed up. `query ix <hash> []` should be for *all* received events, no?

Reply:
Nope. It's correct. You may want to drop some of the events <K according to risks that you may want to take. So without any events you get a version of the database that has been settled.

The first example adds all unsettled events (<K) and then runs the query.